server/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -1133,4 +1133,35 @@ public void test_copy_preserves_the_implied_sub_column_order() throws IOExceptio
                 ")"
             );
     }
+
+    @Test
+    public void test_generated_non_deterministic_value_is_consistent_on_primary_and_replica() throws Exception {
+        execute("""
+            create table tbl (x int, created generated always as now())
+            clustered into 1 shards
+            with (number_of_replicas = 1)
+            """
+        );
+        Path path = tmpFileWithLines(Arrays.asList(
+            "{\"x\": 1}"
+        ));
+        execute("copy tbl from ? with (shared=true)", new Object[] { path.toUri().toString() + "*.json"});
+
+        execute("refresh table tbl");
+        execute("select x, created from tbl");
+
+        // (int) response.rows()[0][0] used to be null because replica
+        // used regular Indexer instead of RawIndexer
+        // with values ["{\"x\": 1}"][some_long_timestamp], ie tried to write String value into int column.
+
+        int x = (int) response.rows()[0][0];
+        long created = (long) response.rows()[0][1];
+
+        // some iterations to ensure it hits both primary and replica
+        for (int i = 0; i < 30; i++) {
```

Comment:
Similar to `TransportSQLActionTest.test_generated_non_deterministic_value_is_consistent_on_primary_and_replica`